flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-hudi/src/main/java/org/apache/flink/cdc/connectors/hudi/sink/function/MultiTableEventStreamWriteFunction.java

```diff
@@ -0,0 +1,745 @@
+            }
+        }
+    }
+
+    @Override
+    public void processElement(Event event, Context ctx, Collector<RowData> out) throws Exception {
+        LOG.debug("Processing event of type: {}", event.getClass().getSimpleName());
+
+        // Route event to appropriate handler based on type
+        if (event instanceof DataChangeEvent) {
+            processDataChange((DataChangeEvent) event, ctx, out);
+        } else if (event instanceof SchemaChangeEvent) {
+            processSchemaChange((SchemaChangeEvent) event);
+        } else if (event instanceof FlushEvent) {
+            processFlush((FlushEvent) event);
+        } else {
+            LOG.warn("Received unknown event type: {}", event.getClass().getName());
+        }
+    }
+
+    /**
+     * Processes schema events. For a {@link CreateTableEvent}, it ensures that the coordinator is
+     * notified and the physical Hudi table is created. For a {@link SchemaChangeEvent}, it updates
+     * the local schema cache.
+     *
+     * <p>Implements {@link EventProcessorFunction#processSchemaChange(SchemaChangeEvent)}.
+     */
+    @Override
+    public void processSchemaChange(SchemaChangeEvent event) throws Exception {
+        TableId tableId = event.tableId();
+        try {
+            if (event instanceof CreateTableEvent) {
+                CreateTableEvent createTableEvent = (CreateTableEvent) event;
+                schemaMaps.put(tableId, createTableEvent.getSchema());
+                LOG.debug("Cached schema for new table: {}", tableId);
+
+                initializedTables.computeIfAbsent(
+                        tableId,
+                        tId -> {
+                            try {
+                                // Send an explicit event to the coordinator so it can prepare
+                                // resources *before* we attempt to write any data.
+                                getOperatorEventGateway()
+                                        .sendEventToCoordinator(
+                                                new CreateTableOperatorEvent(createTableEvent));
+                                LOG.info(
+                                        "Sent CreateTableOperatorEvent to coordinator for new table: {}",
+                                        tId);
+
+                                // Now, create the physical dir for Hudi table.
+                                Configuration tableConfig = createTableSpecificConfig(tId);
+                                createHudiTablePath(tableConfig, tId);
+                            } catch (Exception e) {
+                                // Re-throw to fail the Flink task if initialization fails.
+                                throw new RuntimeException(
+                                        "Failed during first-time initialization for table: " + tId,
+                                        e);
+                            }
```

Comment:
` initializedTables.put(tableId, true);` should be put here instead of inside getOrCreateTableFunction?